grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -235,37 +242,50 @@ class ProductAvailabilityService {
                         FROM location_supported_activities lsa_select
                         GROUP BY lsa_select.location_id
                     ) lsa ON lsa.location_id = l.id
-                WHERE (r.origin_id = :locationId AND r.status IN (${RequisitionStatus.listPending().collect { "'$it'" }.join(',')}))
-                  AND (ri.id IS NOT NULL AND (ii.lot_status IS NULL OR ii.lot_status != 'RECALLED') AND (lsa.activities IS NULL OR lsa.activities NOT LIKE '%HOLD_STOCK%'))
+                WHERE (r.origin_id = :locationId 
+                    AND r.status IN (:pendingRequisitionStatuses))
+                  AND (ri.id IS NOT NULL AND (ii.lot_status IS NULL OR ii.lot_status != 'RECALLED') 
+                  AND (lsa.activities IS NULL OR lsa.activities NOT LIKE '%HOLD_STOCK%'))
                   AND (:productId = '' OR ri.product_id = :productId)
                 GROUP BY pli.bin_location_id, pli.inventory_item_id
                 UNION
                 SELECT
                     pli.bin_location_id as bin_location_id,
                     pli.inventory_item_id as inventory_item_id,
-                    sum(pli.quantity) as quantity
+                    sum(pli.quantity) as quantity_picked
                 FROM picklist_item pli
                     INNER JOIN picklist p ON pli.picklist_id = p.id
                     LEFT JOIN order_item oi ON pli.order_item_id = oi.id
                     LEFT JOIN `order` o ON p.order_id = o.id
                     LEFT JOIN inventory_item ii ON pli.inventory_item_id = ii.id
-                WHERE (o.origin_id = :locationId AND o.status IN (${OrderStatus.listPending().collect { "'$it'" }.join(',')}))
+                WHERE (o.origin_id = :locationId 
+                    AND o.status IN (:pendingOrderStatuses))
                   AND (oi.id IS NOT NULL)
                   AND (:productId = '' OR oi.product_id = :productId)
                 GROUP BY pli.bin_location_id, pli.inventory_item_id
             ) as requisition_order_union
             GROUP BY bin_location_id, inventory_item_id;
         """
 
-        def results = dataService.executeQuery(
-            query,
+        SQLQuery sqlQuery = sessionFactory.currentSession.createSQLQuery(query)
+        List results = sqlQuery.addScalar("bin_location_id", Hibernate.STRING)
+                .addScalar("inventory_item_id", Hibernate.STRING)
+                .addScalar("quantity_allocated", Hibernate.BIG_DECIMAL)
+                .setString("locationId", location?.id)
+                .setString("productId", product?.id ?: '')
+                .setParameterList("pendingRequisitionStatuses", RequisitionStatus.listPending().collect { it.name() })
+                .setParameterList("pendingOrderStatuses", OrderStatus.listPending().collect { it.name() })
```

Comment:
LGTM, just wondering if it would work without `.collect` here, since `toString()` returns `name()`, if Hibernate would convert that properly